### PR TITLE
Feature/10318 refactor els submission

### PR DESCRIPTION
**Pull request by 30mar**
## Description
Refactor ELS submission
Not included:

- 3 commented three unit tests
- OTP submission

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-10320

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Services/ELS LoggingService/__tests__/ELSServiceTests.swift [hunk 1]
```diff
@@ -189,12 +212,11 @@ class ELSServiceTests: CWATestCase {
 		let riskProvider = MockRiskProvider()
 		let otpService = OTPService(
 			store: store,
-			client: client,
+			client: ClientMock(), // might fail
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Services/ELS LoggingService/__tests__/ELSServiceTests.swift:215]: Didn't get the comment

File: src/xcode/ENA/ENA.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -7437,6 +7444,16 @@
 			path = HealthCertifiedPerson;
 			sourceTree = "<group>";
 		};
+		8FE68F9128A1469E003E86E1 /* ELS */ = {
+			isa = PBXGroup;
+			children = (
+				8FE68F9228A146CC003E86E1 /* SubmitELSResource.swift */,
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA.xcodeproj/project.pbxproj:7450]: Can you please rename this and the corresponding files to ELSSubmitResource to align with PPASubmitResource

File: src/xcode/ENA/ENA/Source/HTTPClient/Resources/ELS/SubmitELSResource.swift [hunk 3]
```diff
@@ -0,0 +1,62 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+
+enum SubmitELSResourceError: Error {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/HTTPClient/Resources/ELS/SubmitELSResource.swift:7]:
```suggestion
enum ELSSubmitResourceError: Error {
```

File: src/xcode/ENA/ENA/Source/HTTPClient/Resources/ELS/SubmitELSRecieveModel.swift [hunk 4]
```diff
@@ -0,0 +1,10 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+
+struct SubmitELSReceiveModel: Codable {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/HTTPClient/Resources/ELS/SubmitELSRecieveModel.swift:7]:
```suggestion
struct ELSSubmitReceiveModel: Codable {
```

File: src/xcode/ENA/ENA/Source/HTTPClient/Resources/ELS/SubmitElSSendModel.swift [hunk 5]
```diff
@@ -0,0 +1,12 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+
+struct SubmitElSSendModel: Encodable {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/HTTPClient/Resources/ELS/SubmitElSSendModel.swift:7]:
```suggestion
struct ElSSubmitSendModel: Encodable {
```

**Review by naveeddotio — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Services/ELS LoggingService/__tests__/ELSServiceTests.swift [hunk 6]
```diff
@@ -65,94 +76,103 @@ class ELSServiceTests: CWATestCase {
 		}
 		XCTAssertEqual(ELSError.ppacError(.generationFailed), error)
 	}
+	// TO DO Related to OPT refactoring part
```
Comment (flxschmidt) [inline src/xcode/ENA/ENA/Source/Services/ELS LoggingService/__tests__/ELSServiceTests.swift:79]: Before we merge commented code, we could also disable the dedicated test(s).

**Review by naveeddotio — CHANGES_REQUESTED** [review]
Blocking until the submit els issue is resolved.

**Review by naveeddotio — APPROVED** [review]
Works as expected.

File: src/xcode/ENA/TestPlans/AllTests.xctestplan [hunk 7]
```diff
@@ -78,6 +78,9 @@
     {
       "parallelizable" : true,
       "skippedTests" : [
+        "ELSServiceTests\/testGIVEN_ELSService_WHEN_UploadIsTriggered_THEN_EmptyLogFileIsReturned()",
+        "ELSServiceTests\/testGIVEN_ELSService_WHEN_UploadIsTriggered_THEN_OTPErrorIsReturned()",
+        "ELSServiceTests\/testGIVEN_ELSService_WHEN_UploadIsTriggered_THEN_ServerErrorIsReturned()",
```
Comment (flxschmidt) [inline src/xcode/ENA/TestPlans/AllTests.xctestplan:83]: 👍

**Review by flxschmidt — APPROVED** [review]
LGTM!

**Pull request merged**